File: cap05/python/ahorcadoV1.py
PALABRA = "PAPA" # "ABRACADABRA"

COMODIN = "_"

TITULO = "=== El Ahorcado ==="
DIME = "Dime una letra\n"
ENHORABUENA = "Enhorabuena, has resuelto la palabra buscada:"

print(TITULO)
objetivo = PALABRA
enJuego = COMODIN * len(objetivo) # Para generar "____"
print(enJuego)
adivinado = False
while not adivinado:
    letra = input(DIME)
    for i in range(len(objetivo)):
        if objetivo[i] == letra:
            enJuego = enJuego[:i] + letra + enJuego[i+1:]
    adivinado = COMODIN not in enJuego
    print(enJuego)
print(ENHORABUENA, objetivo)

File: cap05/java/AhorcadoV1.java
import java.util.Scanner;

public class AhorcadoV1 {

	private static final String PALABRA = "PAPA"; // "ABRACADABRA"

	private static final char COMODIN = '_';

	private static final String TITULO = "=== El Ahorcado ===";
	private static final String DIME = "Dime una letra";
	private static final String ENHORABUENA = 
			"Enhorabuena, has resuelto la palabra buscada: ";

	public static void main(String[] args) {
		System.out.println(TITULO);
		String objetivo = PALABRA;
		char[] enJuego = generarBase(objetivo);
		System.out.println(enJuego);
		Scanner s = new Scanner(System.in);
		boolean adivinado;
		do {
			System.out.println(DIME);
			char jugada = s.next().charAt(0); // tomamos la 1ª letra
			for (int i = 0; i < objetivo.length(); i++) {
				char letra = objetivo.charAt(i);
				if (letra == jugada) {
					enJuego[i] = jugada;
				}
			}
			adivinado = true;
			for (int i = 0; i < objetivo.length(); i++) {
				adivinado &= enJuego[i] != COMODIN;
			}
			System.out.println(enJuego);
		} while (!adivinado);
		
		System.out.println(ENHORABUENA + objetivo);
		s.close();
	}
	
	/**
	 * Genera un array de caracteres rellenos con el carácter COMODIN
	 * de la misma longitud que la palabra recibida.
	 * @param palabra de la que tomar la longitud
	 * @return array de COMODIN de la misma longitud que palabra
	 */
	private static char[] generarBase(String palabra) {
		char[] base = new char[palabra.length()];
		for (int i = 0; i < base.length; i++) {
			base[i] = COMODIN;
		}
		return base;
	}
}
